app/src/main/java/org/p2p/wallet/home/addmoney/ui/AddMoneyDialog.kt

```diff
@@ -0,0 +1,154 @@
+        cellAdapter.updateItem<MainCellModel>(
+            predicate = { it is MainCellModel && it.payload == itemType },
+            transform = {
+                val rightSideCellModel = if (showProgress) {
+                    RightSideCellModel.Progress(
+                        indeterminateProgressTint = R.color.night
+                    )
+                } else {
+                    RightSideCellModel.IconWrapper(
+                        iconWrapper = IconWrapperCellModel.SingleIcon(
+                            icon = ImageViewCellModel(
+                                icon = DrawableContainer(R.drawable.ic_chevron_right),
+                                iconTint = R.color.icons_mountain
+                            )
+                        )
+                    )
+                }
+                it.copy(rightSideCellModel = rightSideCellModel)
+            }
+        )
+    }
+
+    override fun navigateToBankTransferTarget(target: StrigaUserStatusDestination) {
+        strigaFragmentFactory.signupFlowFragment(target)?.let(::dismissAndNavigate)
+    }
+
+    override fun navigateToBankCard(tokenToBuy: Token, paymentMethod: PaymentMethod.MethodType) {
+        dismissAndNavigate(
+            NewBuyFragment.create(
+                token = tokenToBuy,
+                preselectedMethodType = paymentMethod
+            )
+        )
+    }
+
+    override fun navigateToKycPending() {
+        dismiss()
+        strigaFragmentFactory.showPendingBottomSheet(requireActivity().supportFragmentManager)
+    }
+
+    override fun setCellItems(items: List<AnyCellItem>) {
+        cellAdapter.items = items
+    }
+
+    override fun navigateToCrypto() {
+        dismissAndNavigate(receiveFragmentFactory.receiveFragment())
+    }
+
+    override fun showUiKitSnackBar(
+        message: String?,
+        messageResId: Int?,
+        onDismissed: () -> Unit,
+        actionButtonResId: Int?,
+        actionBlock: ((Snackbar) -> Unit)?
+    ) {
+        require(message != null || messageResId != null) {
+            "Snackbar text must be set from `message` or `messageResId` params"
+        }
+        val snackbarText: String = message ?: messageResId?.let(::getString)!!
+        val root = requireView().rootView
+        if (actionButtonResId != null && actionBlock != null) {
+            root.showSnackbarShort(
+                snackbarText = snackbarText,
+                actionButtonText = getString(actionButtonResId),
+                actionButtonListener = actionBlock,
+                enableBottomNavOffset = false
+            )
+        } else {
+            root.showSnackbarShort(
+                snackbarText = snackbarText,
+                onDismissed = onDismissed,
+                enableBottomNavOffset = false
+            )
+        }
+    }
+
+    private fun dismissAndNavigate(fragment: Fragment) {
+        replaceFragment(fragment)
+        dismiss()
```

Comment:
`dismissAllowingStateLoss`, otherwise you should catch saved instance state error sometimes